listenbrainz/webserver/views/atom.py

```diff
@@ -0,0 +1,242 @@
+    days = _parse_int_arg("days", DEFAULT_NUMBER_OF_FRESH_RELEASE_DAYS)
+    if days < 1 or days > MAX_NUMBER_OF_FRESH_RELEASE_DAYS:
+        return Response(status=400)
+
+    past = _parse_bool_arg("past", True)
+    future = _parse_bool_arg("future", True)
+
+    try:
+        db_releases, _ = get_sitewide_fresh_releases(ts_conn, date.today(), days, "release_date", past, future)
+    except Exception as e:
+        current_app.logger.error("Server failed to get latest release: {}".format(e))
+        raise Response(status=500)
+
+    fg = FeedGenerator()
+    fg.id(f"https://listenbrainz.org/explore/fresh-releases")
+    fg.title(f"Fresh Releases - ListenBrainz")
+    fg.author({"name": "ListenBrainz"})
+    fg.link(href=f"https://listenbrainz.org/explore/fresh-releases", rel="alternate")
+    fg.link(
+        href=f"https://listenbrainz.org/syndication-feed/fresh_releases",
+        rel="self",
+    )
+    fg.logo("https://listenbrainz.org/static/img/listenbrainz_logo_icon.svg")
+    fg.language("en")
+
+    for r in db_releases:
+        release_name = r.release_name
+        artist_credit_name = r.artist_credit_name
+        time = r.release_date.strftime("%Y-%m-%d")
+
+        fe = fg.add_entry()
+        fe.id(
+            f"https://listenbrainz.org/syndication-feed/fresh_releases/{time}/{artist_credit_name}/{release_name}"
+        )
+        fe.title(f"{release_name} by {artist_credit_name}")
+        
+        _content = render_template("atom/fresh_releases.html",
+                                   artist_credit_name=artist_credit_name,
+                                   release_name=release_name,
+                                   time=time
+        )
+
+        fe.content(
+            content=_content,
+            type="html",
+        )
+        t = datetime.combine(r.release_date, datetime.min.time())
+        fe.published(t.replace(tzinfo=timezone.utc))
+        fe.updated(t.replace(tzinfo=timezone.utc))
+
+    atomfeed = fg.atom_str(pretty=True)
+
+    return Response(atomfeed, mimetype="application/atom+xml")
+
+
+@atom_bp.route("/user/<user_name>/fresh_releases", methods=["GET"])
+@crossdomain
+def get_releases(user_name):
+    """
+    Get fresh releases for a user, sorted by release date.
+
+    :param past: Whether to show releases in the past. Default True.
+    :param future: Whether to show releases in the future. Default True.
+    :statuscode 200: The feed was successfully generated.
+    :statuscode 400: Bad request.
+    :statuscode 404: The user does not exist.
+    :resheader Content-Type: *application/atom+xml*
+    """
+    user = db_user.get_by_mb_id(db_conn, user_name)
+    if user is None:
+        return Response(status=404)
+
+    past = _parse_bool_arg("past", True)
+    future = _parse_bool_arg("future", True)
+
+    data = db_get_fresh_releases(user["id"])
+    releases = data["releases"] if data else []
+    releases = sorted(releases, key=lambda k: k.get("release_date", ""))
+
+    if not past:
```

Comment:
Date is possibly parsed twice here. I would parse it once and then use the result in both places.